### PR TITLE
TM-3680 Update/panel meetings

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/740)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/330)

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
Will do cleanup on Monday

**Comment by elizabeth-jimenez** [comment]
@scott062  @SidelineCory24  Was going to do the sorts in this PR, but currently a bit blocked by dev1 stuff, so this PR is ready for review. Sorts will be addressed in another.

**Comment by SidelineCory24** [comment]
> @scott062 @SidelineCory24 Was going to do the sorts in this PR, but currently a bit blocked by dev1 stuff, so this PR is ready for review. Sorts will be addressed in another.

dev1 needs to get `sorted` out 🥁

File: src/actions/panelMeetingAgendas.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ export function panelMeetingAgendasExport(query = {}) {
 export function panelMeetingAgendasFetchData(query = {}, id) {
   return (dispatch) => {
     const q = convertQueryToString(query);
-    const endpoint = `/fsbid/panel/${id}/agendas`;
+    const endpoint = `/fsbid/panel/${id}/agendas/`;
```
Comment (SidelineCory24) [inline src/actions/panelMeetingAgendas.js:61]: should we close the other pr? or good to keep so both work?

Comment (elizabeth-jimenez) [reply src/actions/panelMeetingAgendas.js:61]: other PR is a dual merge on another PR, which is a dual merge on another 👀

File: src/actions/panelMeetings.js [hunk 2]
```diff
@@ -101,84 +66,107 @@ export function panelMeetingsFetchData(query = {}) {
...
+        batch(() => {
+          dispatch(panelMeetingsFiltersFetchDataErrored(true));
+          dispatch(panelMeetingsFiltersFetchDataLoading(false));
+        });
+      });
+  };
+}
+
+
+export function panelMeetingsExport(query = {}) {
```
Comment (SidelineCory24) [inline src/actions/panelMeetings.js:152]: I think we should disable this in the jsx until the ep is available.

File: src/Constants/PropTypes.js [hunk 3]
```diff
@@ -752,22 +752,26 @@ export const AB_EDIT_DETAILS_OBJECT = PropTypes.shape({
...
-  posNum: PropTypes.string,
-  action: PropTypes.string,
-  travel: PropTypes.string,
-});
-
-export const TEMP_FAKE_DATA = PropTypes.shape({
-  status: PropTypes.string,
-  legs: PropTypes.arrayOf(TEMP_FAKE_LEGS),
-  panelDate: PropTypes.string,
+export const PANEL_MEETING_DATES = PropTypes.shape({
```
Comment (SidelineCory24) [inline src/Constants/PropTypes.js:755]: we prefer the real😎

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 4]
```diff
@@ -26,66 +23,33 @@ const PanelMeetingSearch = ({ isCDO }) => {
...
   const userSelections = useSelector(state => state.panelMeetingsSelections);
 
   const panelMeetings = get(panelMeetings$, 'results') || [];
 
-  // TO-DO: complete integration based off of BE/WS data
   const [limit, setLimit] = useState(get(userSelections, 'limit') || PANEL_MEETINGS_PAGE_SIZES.defaultSize);
   const [ordering, setOrdering] = useState(get(userSelections, 'ordering') || PANEL_MEETINGS_SORT.defaultSort);
 
   const [selectedMeetingType, setSelectedMeetingType] = useState(get(userSelections, 'selectedMeetingType') || []);
-  const [selectedMeetingDate, setSelectedMeetingDate] = useState(get(userSelections, 'selectedMeetingDate') || null);
```
Comment (SidelineCory24) [inline src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:42]: no more panel date?

Comment (SidelineCory24) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:42]: disregard, I see it the updated payload

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 5]
```diff
@@ -95,23 +59,17 @@ const PanelMeetingSearch = ({ isCDO }) => {
...
-
-    // need to set to beginning of the day to avoid timezone issues
-    'pmd-start': isDate(get(selectedMeetingDate, '[0]')) ? startOfDay(get(selectedMeetingDate, '[0]')).toJSON() : '',
-    'pmd-end': isDate(get(selectedMeetingDate, '[1]')) ? startOfDay(get(selectedMeetingDate, '[1]')).toJSON() : '',
-
+    type: selectedMeetingType.map(meetingObject => (get(meetingObject, 'code'))),
+    status: selectedMeetingStatus.map(meetingObject => (get(meetingObject, 'code'))),
     // Free Text
-    q: textInput || textSearch,
+    id: textInput || textSearch,
```
Comment (SidelineCory24) [inline src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:65]: should we display the ID now or will cdo's/ao's have this on hand

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 6]
```diff
@@ -195,150 +151,131 @@ const PanelMeetingSearch = ({ isCDO }) => {
...
     if (panelMeetingsIsLoading) {
-      return <Spinner type="bureau-results" class="homepage-position-results" size="big" />;
-    // } else if (panelMeetingsHasErrored) {
-    // eslint-disable-next-line max-len
-    //   toReturn = <Alert type="error" title="Error loading panel meetings" messages={[{ body: 'Please try again.' }]} />;
+      toReturn = <Spinner type="bureau-results" class="homepage-position-results" size="big" />;
+    } else if (panelMeetingsHasErrored || panelMeetingsFiltersHasErrored) {
+      toReturn = <Alert type="error" title="Error loading panel meetings" messages={[{ body: 'Please try again.' }]} />;
     } else if (noPanelMeetingResults) {
-      return <Alert type="info" title="No results found" messages={[{ body: 'Please broaden your search criteria and try again.' }]} />;
```
Comment (SidelineCory24) [inline src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:217]: image

image

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx [hunk 7]
```diff
@@ -195,150 +151,131 @@ const PanelMeetingSearch = ({ isCDO }) => {
...
+            label="Search for a Panel Meeting"
+            placeHolder="Search using Panel ID"
+          />
+          <div className="filterby-container">
+            <div className="filterby-label">Filter by:</div>
+            <div className="filterby-clear">
+              {clearFilters &&
+                <button className="unstyled-button" onClick={resetFilters}>
+                  <FA name="times" />
+                  Clear Filters
```
Comment (SidelineCory24) [inline src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:199]: filters working as expected

File: src/Components/Panel/PanelMeetingSearchRow/PanelMeetingSearchRow.jsx [hunk 8]
```diff
@@ -1,50 +1,47 @@
...
-          <div className="panel-meeting-row-data-point">
-            <dt>Addendum Cut-Off:</dt>
-            <dd>{addendum_cutoff}</dd>
-          </div>
+          {meetingDates$.map(pmd => {
+            if (get(pmd, 'mdt_desc_text')) {
+              return (
+                <div className="panel-meeting-row-data-point">
+                  <dt>{get(pmd, 'mdt_desc_text')}:</dt>
+                  <dd>{formatDate(get(pmd, 'pmd_dttm'), 'MM/DD/YYYY HH:mm') || FALLBACK}</dd>
```
Comment (SidelineCory24) [inline src/Components/Panel/PanelMeetingSearchRow/PanelMeetingSearchRow.jsx:39]: Should this be `Official Addendum Cutoff` instead to match the others? I know this is more of a WS change and may even need C + R.
image

**Comment by SidelineCory24** [comment]
Even if the panel meeting doesn't have every date yet, is that date field expected to not be shown or should we have a fallback for none listed?
image

image

File: src/Components/Panel/PanelMeetingSearchRow/PanelMeetingSearchRow.jsx (hunk 8 above)
Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingSearchRow/PanelMeetingSearchRow.jsx:39]: 🤷‍♀️

Comment (SidelineCory24) [reply src/Components/Panel/PanelMeetingSearchRow/PanelMeetingSearchRow.jsx:39]: discussed in planning meeting, a ticket was created to remove the word `Official` instead, a ticket has been created and added to the backlog.

**Comment by SidelineCory24** [comment]
> > Even if the panel meeting doesn't have every date yet, is that date field expected to not be shown or should we show that date title and have a fallback for none listed? image
> > image
> 
> Good question, you can bring it up at ~Grooming~ Planning today, along with the other question about the labels. This is not part of the Panels EP ticket - I started to work on the dates and then created a ticket https://metaphase.atlassian.net/browse/TM-3951 to address a bit more of them

PO requested that all dates should show and appear as `None listed` if there is no value. This work will be done on another pr/ticket.

File: src/actions/panelMeetings.js (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/actions/panelMeetings.js:152]: That was already there - I didn't add it, i just moved it around

File: src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx (hunk 4 above)
Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingSearch/PanelMeetingSearch.jsx:42]: Gaslit yourself - yep, no more Panel Meeting Date filter. Not currently possible to filter on nested WS payload items.

**Comment by SidelineCory24** [comment]
verified filters work as expected after prs this was on hold for were merged.

@elizabeth-jimenez those prs were the only thing holding this up correct?

**Comment by elizabeth-jimenez** [comment]
> verified filters work as expected after prs this was on hold for were merged.
> 
> @elizabeth-jimenez those prs were the only thing holding this up correct?

Yep!

**Comment by elizabeth-jimenez** [comment]
> > verified filters work as expected after prs this was on hold for were merged.
> > @elizabeth-jimenez those prs were the only thing holding this up correct?
> 
> Yep!

@SidelineCory24 This good to go?

**Comment by SidelineCory24** [comment]
> > > verified filters work as expected after prs this was on hold for were merged.
> > > @elizabeth-jimenez those prs were the only thing holding this up correct?
> > 
> > 
> > Yep!
> 
> @SidelineCory24 This good to go?

yes!

**Review by SidelineCory24 — APPROVED** [review]
Looks good to go!🚀

**Pull request merged**